gui-ext.js

```diff
@@ -39,8 +39,12 @@ ProjectDialogMorph.prototype.openProject = function () {
         this.ide.updateUrlQueryString(proj.name, false, true);
     } else if (this.source === 'cloud-shared'){
         this.destroy();
+        var loadingMsg = this.ide.showMessage('Loading project.. ');
         SnapCloud.callService('joinActiveProject', function(response) {
             myself.ide.rawLoadCloudProject(response[0], proj.Public);
+            // since there is no simple way to pass a callback to openProject,
+            // set a small timeout so that the next line is delayed by the blocking deserialization
+            setTimeout(() => loadingMsg.destroy(), 500);
```

Comment:
`showMessage` also accepts a time limit. It might be nicer to simply use that instead of manually destroying the message.

Reply:
That does kill the message after a timeout right? That is better

Reply:
Then what about `myself.ide.showMessage('Loading project.. ', 2);` on line 47 instead?

Reply:
the ideal case would be to have a callback or an event dispatched when the project is done loading and is deserialized. Initially, I proposed manually destroying the message to stay closer to the best case scenario but since there is still a race situation we can just add that 2 second delay to the initial message.